### DISCUSSION TITLE
"Shadow class as WebDriver" feature

### DISCUSSION THREAD

**Discussion by D4VOS** [sukgu/shadow-automation-selenium#50] [Polls]
Implementing this library as a WebDriver allows e.g. to use `@FindBy` in Page Objects (you don't need `@FindElementBy`). 
With such a WebDriver, it won't matter if you automate shadow DOM website or regular DOM.

The only change to the code will be redefining your WebDriver with `driver = new ShadowDriver(driver);`

**Comment by dleszczynski19** [comment]
Good idea

**Comment by AdrianCzarnowski** [comment]
code will be shorter

**Comment by D4VOS** [comment]
As had to use this feature across few projects, I decided to make [own project](https://github.com/D4VOS/shadow-webdriver) which you can get by maven. Enjoy :)
